pkg/gather/common/common.go

```diff
@@ -0,0 +1,14 @@
+package common
+
+import (
+	"context"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatheringClosure is a struct containing a closure each gatherer returns
+// it also contains CanFail field showing if we should just log the failures
+type GatheringClosure struct {
+	Run     func(context.Context) ([]record.Record, []error)
+	CanFail bool
+}
```

Comment:
I am still puzzled by the fact that this is very similar to the `gatheringFunction` type. I think it would be nice to have only one type (but it would probably mean to somehow share the `Gatherer` type), but not sure if it's feasible.

Reply:
Hmm it's probably not a good idea to try to reuse one `Gatherer` type.... :-/

Reply:
Yeah, gatherers have different types so we have to hide it from the calling code somehow and using the same type would make gatherers not flexible.